wandb/util.py

```diff
@@ -891,6 +908,12 @@ def no_retry_auth(e: Any) -> bool:
     # Don't retry bad request errors; raise immediately
     if e.response.status_code in (400, 409):
         return False
+    if e.response.status_code == 404:
+        # Don't retry 404 errors, raise 404 error if it has a non-empty error message
+        # Otherwise trickle down to the default error handling below
```

Comment:
This comment is hard to understand for me. Why does this logic only apply to 404, and not to 401 or 403?